src/moderation/moderation.service.integration.test.ts

```diff
@@ -177,5 +177,38 @@ describe('ModerationService', () => {
 
             expect(charlieCanInteractWithAlice).toBe(true);
         });
+
+        it('should prevent interaction when the domain is blocked', async () => {
+            const [[aliceAccount], [bobAccount, bobSite]] = await Promise.all([
+                fixtureManager.createInternalAccount(),
+                fixtureManager.createInternalAccount(null, 'blocked.com'),
+            ]);
+
+            const [charlieAccount] =
+                await fixtureManager.createInternalAccount(bobSite);
+
+            const bobDomain = bobAccount.apId.hostname;
+
+            await client('domain_blocks').insert({
+                blocker_id: aliceAccount.id,
+                domain: bobDomain,
+            });
```

Comment:
Could use the method you added to the fixture manager?

Reply:
Ah yes, that wasn't merged when I wrote this, but I'll do that